xml/hwbp_registers.xml

```diff
@@ -289,4 +292,85 @@
             Other values are reserved for future use.
         </field>
     </register>
+
+    <register name="Exception Trigger" short="etrigger" address="0x7a1">
+        This register is accessible as \Rtdataone when \Ftype is 4.
+
+        If implemented, it allows triggering on a variety of interrupt and
+        exception sources. If such a trigger is taken, all the usual CSRs as
+        well as \Rpc are updated, and the requested action is taken just before
+        the first instruction of the interrupt/exception handler is executed.
+
+        <field name="type" bits="XLEN-1:XLEN-4" access="R" reset="2" />
+        <field name="dmode" bits="XLEN-5" access="R/W" reset="0" />
+        <field name="0" bits="XLEN-6:24" access="R" reset="0" />
+        <field name="source" bits="23:10" access="R/W" reset="0">
+            For each bit that is set, the corresponding interrupt/exception
+            will cause this trigger to fire if bit corresponding to the current
+            mode is also set.
+
+            0: Software interrupt
```

Comment:
I think we need to keep this parallel to the actual $mip/$mie/$mdeleg registers. Can we use \tdatatwo as the mask?

The ISA spec allows additional interrupt sources, and maybe they'll shove something in between some bits that are currently reserved, it seems brittle to re-assign the bit numbers here. Then we can be more generic and say "the coresponding bit in *ip *ie"

Reply:
And perhaps we just want two separate trigger types, one for interrupt vs exception, or maybe we have bits in this register that say whether the mask applies to exceptions and/or interrupts

Reply:
I started out with that, but then it turned out a good portion of the ones in *cause are "... from user mode", "... from machine mode" etc. Since we already have S/U/M bits that seemed like unnecessary duplication, and then we need extra language to explain that "external interrupt from user mode" won't actually cause a trigger to fire unless the U bit is also set. And there's a break exception which seems pointless to have here so I excluded it completely.

We could put all the bits in data2, using the low 16 bits for exceptions and high 16 bits for interrupts. Or even put interrupts in data3.

What do you think?

Reply:
I think the "mode" in *cause are misleading, as noted by frequent questions about it on the mailing list. For example "Supervisor software interrupt" vs "Machine software interrupt" are sort of conventional naming that allow SW to route and delegate things a certain way, not something that the hardware really knows the difference between (it just knows they are different).

Reply:
The only one I saw that was explicitly "from X-Mode" is "environment call". 

In general I think we should just keep the parallel structure to avoid having to track changes to the priv spec here as things are potentially added and removed.

Reply:
And obviously some bits would be WARL if there is no code defined for that, which again parallels the ISA spec.

Reply:
There are also the 3 interrupts (software, timer, external), each of which explicitly have from M/S/U mode.

Reply:
That's what I'm saying, that's really just a naming convention. There is no enforcement in the spec that those thigns really happen in M/S/U mode.